compiler/rustc_incremental/src/persist/fs.rs

```diff
@@ -585,23 +585,17 @@ fn extract_timestamp_from_session_dir(directory_name: &str) -> Result<SystemTime
 
 fn timestamp_to_string(timestamp: SystemTime) -> BaseNString {
     let duration = timestamp.duration_since(UNIX_EPOCH).unwrap();
-    let micros = duration.as_secs() * 1_000_000 + (duration.subsec_nanos() as u64) / 1000;
+    let micros: u64 = duration.as_micros().try_into().unwrap();
```

Comment:
I've made conversions explicitly fallible here and other places.

Reply:
Why are we using `u64` here? `to_base_fixed_len` seems to support `u128` too.

Reply:
```rust
    println!("{}", 1234567_u64.to_base_fixed_len(CASE_INSENSITIVE));
    println!("{}", 1234567_u128.to_base_fixed_len(CASE_INSENSITIVE));
```

will give
```
000000000qglj
000000000000000000000qglj
```

so this will change timestamp part of filename making it longer, idk if that needed?

Reply:
hm... i guess if we never expect it to overflow u64 in practice then it's fine. 2^64 microseconds seems to be 584 thousand years...